src/commandline_options.cpp

```diff
@@ -595,18 +598,18 @@ void commandline_options::parse_resolution_ ( const std::string& resolution_stri
 		throw bad_commandline_resolution(resolution_string);
 	}
 
-	resolution = std::pair(xres, yres);
+	resolution.emplace(xres, yres);
```

Comment:
While the new code is nice, the old code seems more readable to me - it's clear that it's replacing the old value (if any) rather than appending to a list. For code that's only called when parsing command line options, my preference is the old code.

Reply:
You make a point, and I respect your opinion, but I have to disagree, here:s my reasoning

To me emplace is a much better api than operator=, in this case, as it show the intent of constructing in place rather than creating a temp pair and then putting it inside the optional.

Maybe emplace is a bad choice for a name...

Again, will revert the change

Reply:
done